### PR TITLE
Google analytics patron report

### LINKED ISSUE #4031: Send an event to Google Analytics when a resource is charged via CDL
# Implementation Notes

Send the following data:

Event Category: `CDL - {patrongroup}`
Event Action: `charge`
Event Label: `{bibid}`

The following CURL is an example of the kind of data we should send:

```
curl --verbose -d "v=1&tid=UA-15870237-29&cid=96afa471-e575-4aac-b07b-f3af862cd9a0&t=event&ec=CDLTest - Staff&ea=charge&el=123456&ua=Figgy" -X POST https://www.google-analytics.com/collect
```

We can get the user's patron group from `https://bibdata.princeton.edu/patron/{{netid}}`. Figgy's already been added to the allowed list to access this data.

### PR DESCRIPTION
**Pull request by christinach**
closes #4031

### CODE REVIEW

File: app/views/viewer/cdl_checkout.html.erb [hunk 1]
```diff
@@ -17,3 +17,4 @@
     <b>This item is currently checked out. The estimated wait time is <%= @charge_manager.estimated_wait_time %>.</b>
   <% end %>
 </div>
+
```
Comment (eliotjordan) [inline app/views/viewer/cdl_checkout.html.erb:20]: issue: extra line

**Comment by eliotjordan** [comment]
Additional specs needed for CDL::EventLogging module:

https://28147-98223070-gh.circle-artifacts.com/0/coverage/index.html#04cc20d054f26ec397c7eac984bc85e4bcfeaa2b

File: app/resources/cdl/event_logging.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  class EventLogging
+    class << self
+      def get_patron_group(netid:)
+        return false unless netid
+        patron_record = Faraday.get "#{CDL::EligibleItemService.bibdata_base}patron/#{netid}"
+        if patron_record.status == 200
+          patron = JSON.parse(patron_record.body).with_indifferent_access
+          patron["patron_group"]
+        else
+          false
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/resources/cdl/event_logging.rb:13]: Oh I didn't consider that it'd log false. I wonder if we should do `netid` instead?

[the remark this answers is not in the document]
Comment (christinach) [reply app/resources/cdl/event_logging.rb:13]: If we add the `netid` then the event will have the `source_metadata_identifier` and the `netid` that requested it. But we wouldn't want to log the activity on this level. 
Would it be better to add in the rails log the user and the status same as https://github.com/pulibrary/orangelight/blob/b6b86181d2df46ada8ec5e4e2fe74517de2eb41b/app/services/bibdata.rb#L7 and also notify the user to contact the admin ??

[the remark this answers is not in the document]
Comment (tpendragon) [reply app/resources/cdl/event_logging.rb:13]: Oh of course, not sure what I was thinking. I think we should probably just return nil like @eliotjordan recommended.

**Comment by christinach** [comment]
@eliotjordan can you check again?

File: app/resources/cdl/event_logging.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    class << self
+      def get_patron_group(netid:)
+        return false unless netid
+        patron_record = Faraday.get "#{CDL::EligibleItemService.bibdata_base}patron/#{netid}"
+        if patron_record.status == 200
+          patron = JSON.parse(patron_record.body).with_indifferent_access
+          patron["patron_group"]
+        end
+      rescue
+        false
```
Comment (eliotjordan) [inline app/resources/cdl/event_logging.rb:14]: I haven't seen the coverage yet, but I think this line is still uncovered. Also, the rescue should also return `nil` as per the previous conversation.

Comment (christinach) [reply app/resources/cdl/event_logging.rb:14]: the line is covered but for `false.` I will update for nil

**Pull request merged**